### ISSUE TITLE
configure coordinate and GeoServer 

### ISSUE THREAD

**Issue by scially** [locationtech/geowave#1360]
I have a shapefile whose coordinates are the CGCS2000 coordinate system. and I use the following command to ingest it into geowave
```
geowave  config addstore load--gwNamespace geowave.load-t accumulo  --zookeeper localhost:2181 --instance accumulo --user admin --password admin
# add index and use CGCS2000
geowave  config addindex -t spatial -c EPSG:4490 load-spatial   
# ingest
geowave ingest localtogw -f geotools-vector /home/hwang/geowave_shp/load/ loadload-spatial
# published to the geoserver
geowave gs addlayer load
```
 GeoServer doesn't display this feature layer, but when i set coordinate system to WGS84, success, and i need help....

**Comment by rfecher** [comment]
That seems strange, it _should_ work.  We may look at that when we have a chance because it seems like it should work.  But alternatively you can use the default WGS84 within GeoWave and that should still work, it'll just reproject the shapefile on ingest.

**Comment by WangSmallHui** [comment]
The coordinates used in my shapef file are CGCS2000，and do you mean that GeoWave will convert the coordinates in my shapefile to WGS84? Or directly use my shape file coordinates as WGS84？Does GeoWave allow me to customize the coordinate system? Thank you！

**Comment by WangSmallHui** [comment]
Because CGCS2000 and WGS84 still have small differences，i want to use CGCS2000, thank you...

**Comment by rfecher** [comment]
Yes, the `-c` option should work as you have tried, but to clarify the alternative I mentioned, GeoWave will convert whatever the coordinate system of the input source data into the coordinate system of the index (so if it is the default WGS84, and the shapefile is CGCS2000, it'll convert to WGS84 on ingest).  Within GeoServer you can also specify the requested coordinate system and GeoWave will convert from the indexed CRS to the requested CRS.

**Comment by WangSmallHui** [comment]
So how does GeoWave know that my shape file is CGCS2000? Is GeoWave determining the shape coordinate system based on the prj file? if i don't providethe prj file？How can it determine CRS？
Is the -c option used to specify the coordinate system of the shapefile? thank you very much...

**Comment by rfecher** [comment]
if you don't have a prj file it has no way of knowing and I believe just assumes WGS84.

**Comment by WangSmallHui** [comment]
Ok, Very reasonable... if i give a prj of CGCS2000, and set -c option to CGCS2000, So is GeoWave not doing any processing?  thanks...

**Comment by WangSmallHui** [comment]
Ok, Very reasonable... if i give a prj of CGCS2000, and set -c option to CGCS2000, So is GeoWave not doing any processing on ingest? thanks...

**Comment by WangSmallHui** [comment]
thank you... I've solved the problem, The table name in accumulo is inconsistent with the name of the table to be queried by the geoserver,I will fix this problem and pull request. Now, i have a new problem... And Does GeoWave support Gaussian projection plane coordinate system, such as "CGCS2000 / 3-degree Gauss-Kruger CM 114E"( EPSG:4547)?

**Comment by rfecher** [comment]
I believe it should support EPSG:4547.  We use GeoTools referencing libraries to resolve coordinate reference systems as well as to apply math transforms.  Is this resolved and can I close it?

**Comment by scially** [comment]
I will try it now, give you feedback today. thank you very much. If it can be ingest successfully, I will close it...

**Comment by scially** [comment]
sorry to bother you, this proble can't be solved successfully.. 
This is my properties file:
![_20180917202009](https://user-images.githubusercontent.com/22127608/45622584-16da8000-bab7-11e8-84d6-e994c5e154c6.png)
And this is layer in geoserver:
![2](https://user-images.githubusercontent.com/22127608/45622662-5012f000-bab7-11e8-8453-6f243a6ecba6.png)
![4](https://user-images.githubusercontent.com/22127608/45622777-c0ba0c80-bab7-11e8-9eec-9454141b5a6f.png)
This is the coordinates displayed in ArcGIS
![3](https://user-images.githubusercontent.com/22127608/45622708-80f32500-bab7-11e8-926c-bfd7af322d09.png)
They are obviously inconsistent...
I want to ask you for advice.Thank you...

**Comment by scially** [comment]
In addtition, I set geowave.charset and it has been loaded into the geowave, but the shapefile is still garbled.
![image](https://user-images.githubusercontent.com/22127608/45622928-43db6280-bab8-11e8-802f-14878f342092.png)
The program will have a warning：
![image](https://user-images.githubusercontent.com/22127608/45622957-52c21500-bab8-11e8-89c6-eeb57467f37d.png)
I want to ask you for advice.Thank you...

**Comment by scially** [comment]
And when I exec 'geowave gs addlayer name', the program will have a error and exit：
![image](https://user-images.githubusercontent.com/22127608/45623004-784f1e80-bab8-11e8-9ef3-4a715c49b04e.png)
 But when I exec 'geowave gs addlayer name' again, the program was executed successfully.

**Comment by scially** [comment]
re:Sorry, I didn't describe it clearly. it is a multipolygon, the coordinate is just a point of this polygon. The ycoordinate in geowave is inconsistent with the  coordinate of ArcGIS...

re: I understand. Do I need to set"-Dgeowave.charset=gbk" to the geoserver configuration file as well?

re: hbase 1.2.0-cdh5.14.4

thanks...

**Comment by rfecher** [comment]
what I meant is, the values for the coordinates in this multipolygon are all the same - is that expected?  Do you know what the value is supposed to be?

Yes, GeoServer will have to also be run with that java property.

Are you installing the geowave's cloudera RPMs or jars?  Such as [this](http://s3.amazonaws.com/geowave-rpms/release/noarch/geowave-0.9.8-cdh5-hbase.201809141204.noarch.rpm) or [this](http://s3.amazonaws.com/geowave-rpms/release-jars/JAR/geowave-hbase-0.9.8-cdh5.jar)

**Comment by scially** [comment]
re: i know, they are different
re: ok. I will try tomorrow.
re: e... i build from source, and mvn install -P cloudera, because the program is not very stable now, so I decided to debug

**Comment by rfecher** [comment]
yeah, `-P cloudera` should suffice, with hopefully `hbase-container-singlejar` `geowave-tools-singlejar` and `geotools-container-singlejar` for the distributed hbase lib, geowave tools, and geoserver plugin resepectively....

in this case it looks like the problem is you had built geowave without the cloudera profile and then didn't clean it when you did apply it.  `setAttribute` returns `void` for cloudera and apache returns `Scan`: https://github.com/cloudera/hbase/blob/cdh5.14.0-release/hbase-client/src/main/java/org/apache/hadoop/hbase/client/OperationWithAttributes.java#L41

so it compiles with a class that was built using apache but it will not be runtime compatible.

Basically, just run `mvn clean` when you're building the jars with `-P cloudera` to safely avoid this problem.

**Comment by scially** [comment]
Ok, i will try...

**Comment by rfecher** [comment]
This issue hasn't been updated in awhile, I think it may be closed.
